### PR TITLE
feat(pipelineexpressions): Adding a v2 expressions evaluator

### PR DESCRIPTION
**Pull request by jeyrschabu**
- Extracted expressions evaluation to new package
- Exposing expression failures via a detailed summary
- Added support for flagging a pipeline to be processed with v2

### CODE REVIEW

**Comment by jeyrschabu** [comment]
The goal of this PR is to introduce a v2 for expression evaluation.
This change has three parts: 
  
1. Expose localized evalution failures:
In addition to error logging in case of a failure, a block like this will appear in the stage execution json
```
"expressionEvaluationSummary": {
  "{#stage('Wait')['context']['test'] == 3": [
    {
      "description": "Expression '{#stage( #root.execution, 'Wait')['context']['test'] == 3' @ 0: No ending suffix '}' for expression starting at character 0: {#stage( #root.execution, 'Wait')['context']['test'] == 3",
      "exceptionType": "org.springframework.expression.ParseException",
      "level": "ERROR",
      "timestamp": 1503293885728
    }
  ]
},
```
 A map where the key is the failed expression and the value is detailed information about the failure

 2. Add support for flagging a pipeline to use the v2 evaluator
 add this `"spel-evaluator" : "v2"` to the pipeline json

 3. Refactor by moving expressions related code to a new package

 Next, lining up a PR that will get rid of the multipass evaluation approach by moving the evaluation at `StageStart`. This will remove the ambiguity on when a given expression is processed. Finally, looking to improve UI support to provide feedback on syntax errors

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionsSupport.java [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+  public static Boolean toBoolean(String str) {
+    return Boolean.valueOf(str);
+  }
+
+  /**
+   * Encodes a string to base64
+   * @param text plain string
+   * @return converted string
+   */
+
```
Comment (robzienert) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionsSupport.java]: These extra newlines bother me.

Comment (jeyrschabu) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionsSupport.java]: Curious to know why? I don't have a problem removing them.

Comment (robzienert) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionsSupport.java]: Primarily from a consistency perspective: It's very atypical; certainly not something we do anywhere else in our projects.

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionTransform.groovy [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      T result
+      Expression exp
+      String escapedExpressionString = null
+      Throwable exception = null
+      try {
+        exp = parser.parseExpression(literalExpression, parserContext)
+        escapedExpressionString = escapeExpression(exp)
+        result = exp.getValue(evaluationContext) as T
+      } catch (Exception e) {
+        log.info("Failed to resolve $source, returning raw value {}", e?.getMessage())
```
Comment (robzienert) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionTransform.groovy]: I don't think there's a condition where `e` would actually be null here?

Comment (jeyrschabu) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionTransform.groovy]: leftover. removing that

**Pull request merged**